.github/workflows/publish.yaml

```
@@ -8,19 +8,28 @@ jobs:
   build:
     name: Build distribution package
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
         with:
           persist-credentials: false
+      - name: Fetch tags
+        run: git fetch --prune --unshallow --tags
```

Although, I just tend to build this right into the checkout step:
```suggestion
      - uses: actions/checkout@v4
        with:
          fetch-depth: >-
            ${{
              github.ref_type == 'tag'
              && 1
              || 0
            }}
          persist-credentials: false
```

---

> Smart, it would certainly work, but isn't more lightweight to only `git fetch --prune --unshallow --tags` (I think it fetches all ancestors of the current branch + all tags), rather to pull all history for _all branches_ and tags (that's what `fetch-depth: 0` does)?

It might be a little more efficient. I don't remember if `--unshallow` would get other branches or not.

> By the way, it would be better to use `fetch-tags: true` in `actions/checkout@v4`, but I tried and it didn't work. According to the internet it might be currently bugged.

I don't think having tags detached from the tree would work. They have to be grafted and exist in the context. `git describe --tags` would need all the commits between `FETCH_HEAD` and the last tag to (1) count them, but also to know which of those tags is closest on the tree. And whether those tags are reachable from `FETCH_HEAD`. Timestamps mean nothing, it should be a tag of the branch. Imagine there's a tag on some other branch — that wouldn't be what `git-describe` finds. Similarly, imagine there's a tag on the very first initial commit in the repo that is dated 2099 — it also won't be what `git-describe` finds.

---

> I don't think having tags detached from the tree would work. They have to be grafted and exist in the context. `git describe --tags` would need all the commits between `FETCH_HEAD` and the last tag to (1) count them, but also to know which of those tags is closest on the tree. And whether those tags are reachable from `FETCH_HEAD`. Timestamps mean nothing, it should be a tag of the branch. Imagine there's a tag on some other branch — that wouldn't be what `git-describe` finds. Similarly, imagine there's a tag on the very first initial commit in the repo that is dated 2099 — it also won't be what `git-describe` finds.

:+1: